### ISSUE TITLE
[addon-docs] TS code generation for a story doesn't render if a JSX element is typed

### ISSUE THREAD

**Issue by lgraziani2712** [storybookjs/storybook#8827]
**Describe the bug**
Typing a JSX element will prevent the documentation to render the story code.

**To Reproduce**

https://github.com/lgraziani2712/sb-8827

1. Create a story in a TSX file.
2. Type a generic JSX element.
3. Open storybook and see the error.

**Expected behavior**
To render the story code the same way as if the JSX element wouldn't have been typed.

**Code snippets**

```tsx
// Wrapper

interface Props<P> {
  initial?: P;
  children(val: P): JSX.Element;
}

function Wrapper<P>({ initial, children }: Props<P>) {
  const [state, setState] = useState(initial || null);

  return children({state, setState });
}

// Story
import Component, { ComponentProps } from '.';

export const someStory = () => (
  // Since I'm not setting a default value, TS doesn't know the type.
  // Typing the element makes TS stop complaining but the story code doesn't render.
  // If I delete the typing, the code starts rendering but TS complains.
  <Wrapper<ComponentProps>>
    {(props) => (<Component {...props} />)}
  </Wrapper>
);
```

**System:**
```
  System:
    OS: Linux 4.15 Ubuntu 18.04.3 LTS (Bionic Beaver)
    CPU: (4) x64 Intel(R) Core(TM) i7-4510U CPU @ 2.00GHz
  Binaries:
    Node: 10.16.0 - ~/.nvm/versions/node/v10.16.0/bin/node
    Yarn: 1.19.1 - /usr/bin/yarn
    npm: 6.9.0 - ~/.nvm/versions/node/v10.16.0/bin/npm
  Browsers:
    Chrome: 78.0.3904.97
    Firefox: 71.0b8
  npmPackages:
    @storybook/addon-actions: 5.2.5 => 5.2.5 
    @storybook/addon-docs: 5.2.5 => 5.2.5 
    @storybook/addon-knobs: 5.2.5 => 5.2.5 
    @storybook/preset-typescript: 1.1.0 => 1.1.0 
    @storybook/react: 5.2.5 => 5.2.5
```

**Comment by shilman** [comment]
Can you provide a repro repo for this?

**Comment by lgraziani2712** [comment]
Sure!

**Comment by lgraziani2712** [comment]
Here it is! https://github.com/lgraziani2712/sb-8827

**Comment by lgraziani2712** [comment]
This is still a bug

**Comment by shilman** [comment]
We've just released zero-config typescript support in 6.0-beta. Please upgrade and test it!

- [Release 6.0.0-beta.12+](https://github.com/storybookjs/storybook/releases/tag/v6.0.0-beta.12)
- [Migration instructions](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#zero-config-typescript)
- [Updated docs](https://github.com/storybookjs/storybook/blob/next/docs/src/pages/configurations/typescript-config/index.md)

Thanks for your help and support getting this stable for release!

**Comment by lgraziani2712** [comment]
I can confirm this is partially fixed in v6-beta (I'm using beta-25). It's partially fixed because although it renders the code now, most of the code is being grayed out:

Specifying the generic:
![image](https://user-images.githubusercontent.com/15050200/84405659-d56cfb00-abde-11ea-8b81-a6f1f3c91be0.png)

Without specifying it:
![image](https://user-images.githubusercontent.com/15050200/84405735-eddd1580-abde-11ea-97b1-e6e46dd36c90.png)
